src/main/webapp/app/home/home.component.ts

```diff
@@ -91,8 +98,18 @@ export class HomeComponent implements OnInit, AfterViewInit {
         });
     }
 
-    ngAfterViewInit() {
-        this.renderer.selectRootElement('#username', true).focus();
+    ngAfterViewChecked() {
+        // Only focus the username input once, not on every update
+        if (this.inputFocused || this.loading) {
```

Comment:
Just a question: Do you really need to add the `inputFocused` variable or is it redundant because if the page isn't loaded, you shouldn't focus input?

Reply:
`inputFocused` is needed for after the page is actually visible, because losing the focus of the field triggers an update which triggers `AfterViewChecked` again, which then in turn focuses the field again, getting this stuck. We only care about focusing the input field once after the loading is done.